### PR TITLE
[Neo Plugin UT] Rpcserver unit test on node

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This is the second pr that focus on adding unit tests for rpcservers. 

Fixes # https://github.com/neo-project/neo/issues/3350

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Here is the list of method names extracted from the provided code:

1. TestGetVersion
2. TestSendRawTransaction_Normal
3. TestSendRawTransaction_InvalidTransactionFormat
4. TestSendRawTransaction_InsufficientBalance
5. TestSendRawTransaction_InvalidSignature
6. TestSendRawTransaction_InvalidScript
7. TestSendRawTransaction_InvalidAttribute
8. TestSendRawTransaction_Oversized
9. TestSendRawTransaction_Expired
10. TestSendRawTransaction_PolicyFailed
11. TestSendRawTransaction_AlreadyInPool
12. TestSendRawTransaction_AlreadyInBlockchain
13. TestSendRawTransaction_MemoryPoolFull
14. TestSubmitBlock_Normal
15. TestSubmitBlock_InvalidBlockFormat
16. TestSubmitBlock_AlreadyExists
17. TestSubmitBlock_InvalidBlock
18. TestSendRawTransaction_NullInput
19. TestSendRawTransaction_EmptyInput
20. TestSubmitBlock_NullInput
21. TestSubmitBlock_EmptyInput

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by vncoelho** [comment]
Coverage is not working and some verification are failing, Jimmy.

**Comment by Jim8y** [comment]
> Coverage is not working and some verification are failing, Jimmy.

i.know, i will fix that, i changed testing utils and that caused some test fail

**Comment by vncoelho** [comment]
> > Coverage is not working and some verification are failing, Jimmy.
> 
> i.know, i will fix that, i changed testing utils and that caused some test fail

Sure, as soon as fixed I will review here again.

I just verified something related to Debug for the method clear of mempool.
What is the real use of this on the UTs themself?

**Comment by Jim8y** [comment]
@cschuchardt88 @shargon @superboyiii

**Comment by Jim8y** [comment]
Have to update some existing tests cause i updated the timestamp of the test block since it was the same as genesis block.

**Review by cschuchardt88 — DISMISSED** [review]

File: tests/Neo.UnitTests/TestUtils.Transaction.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Neo.UnitTests;
+
+public partial class TestUtils
+{
+
+    public static Transaction CreateInvalidTransaction(DataCache snapshot, NEP6Wallet wallet, WalletAccount account, InvalidTransactionType type, UInt256 conflict = null)
```
Comment (shargon) [inline tests/Neo.UnitTests/TestUtils.Transaction.cs:31]:
```suggestion
    public static Transaction CreateInvalidTransaction(DataCache snapshot, NEP6Wallet wallet, WalletAccount account, InvalidTransactionType type, UInt256 conflict = null)
```

File: src/Neo/Ledger/MemoryPool.cs [hunk 2]
```diff
@@ -659,5 +659,26 @@ internal bool ReVerifyTopUnverifiedTransactionsIfNeeded(int maxToVerify, DataCac
 
             return _unverifiedTransactions.Count > 0;
         }
+
+#if DEBUG
+        // This method is only for test purpose
+        // Do not remove it from the DEBUG build
+        internal void Clear()
```
Comment (shargon) [inline src/Neo/Ledger/MemoryPool.cs:666]: I think it doesn't hurt without debug, but it's ok for me

**Review by shargon — APPROVED** [review]

**Review by cschuchardt88 — APPROVED** [review]

**Comment by Jim8y** [comment]
@superboyiii

File: src/Neo/Ledger/MemoryPool.cs (hunk 2 above)
Comment (shargon) [reply src/Neo/Ledger/MemoryPool.cs:666]: If we compile in release maybe it fault because the UT require this method

Comment (cschuchardt88) [reply src/Neo/Ledger/MemoryPool.cs:666]: when add to test the same way as `#if DEBUG`

Comment (shargon) [reply src/Neo/Ledger/MemoryPool.cs:666]: @Jim8y we need to compile the whole sln in release, I think it won't work with this change

**Pull request merged**